// wangjunfui/dist/wangjunfui.js
function Zt(e, t) {
  const n = /* @__PURE__ */ Object.create(null), s = e.split(",");
  for (let r = 0; r < s.length; r++)
    n[s[r]] = !0;
  return t ? (r) => !!n[r.toLowerCase()] : (r) => !!n[r];
}
function je(e) {
  if (h(e)) {
    const t = {};
    for (let n = 0; n < e.length; n++) {
      const s = e[n], r = C(s) ? nn(s) : je(s);
      if (r)
        for (const o in r)
          t[o] = r[o];
    }
    return t;
  } else {
    if (C(e))
      return e;
    if (V(e))
      return e;
  }
}
const kt = /;(?![^(]*\))/g, en = /:([^]+)/, tn = /\/\*.*?\*\//gs;
function nn(e) {
  const t = {};
  return e.replace(tn, "").split(kt).forEach((n) => {
    if (n) {
      const s = n.split(en);
      s.length > 1 && (t[s[0].trim()] = s[1].trim());
    }
  }), t;
}
function be(e) {
  let t = "";
  if (C(e))
    t = e;
  else if (h(e))
    for (let n = 0; n < e.length; n++) {
      const s = be(e[n]);
      s && (t += s + " ");
    }
  else if (V(e))
    for (const n in e)
      e[n] && (t += n + " ");
  return t.trim();
}
const P = process.env.NODE_ENV !== "production" ? Object.freeze({}) : {}, rn = process.env.NODE_ENV !== "production" ? Object.freeze([]) : [], ze = () => {
}, sn = /^on[^a-z]/, on = (e) => sn.test(e), $ = Object.assign, cn = (e, t) => {
  const n = e.indexOf(t);
  n > -1 && e.splice(n, 1);
}, ln = Object.prototype.hasOwnProperty, g = (e, t) => ln.call(e, t), h = Array.isArray, Y = (e) => Oe(e) === "[object Map]", un = (e) => Oe(e) === "[object Set]", w = (e) => typeof e == "function", C = (e) => typeof e == "string", Ke = (e) => typeof e == "symbol", V = (e) => e !== null && typeof e == "object", an = (e) => V(e) && w(e.then) && w(e.catch), fn = Object.prototype.toString, Oe = (e) => fn.call(e), dt = (e) => Oe(e).slice(8, -1), pn = (e) => Oe(e) === "[object Object]", He = (e) => C(e) && e !== "NaN" && e[0] !== "-" && "" + parseInt(e, 10) === e, dn = (e) => {
  const t = /* @__PURE__ */ Object.create(null);
  return (n) => t[n] || (t[n] = e(n));
}, hn = dn((e) => e.charAt(0).toUpperCase() + e.slice(1)), ge = (e, t) => !Object.is(e, t), _n = (e, t, n) => {
  Object.defineProperty(e, t, {
    configurable: !0,
    enumerable: !1,
    value: n
  });
}, gn = (e) => {
  const t = parseFloat(e);
  return isNaN(t) ? e : t;
};
let ke;
const mn = () => ke || (ke = typeof globalThis < "u" ? globalThis : typeof self < "u" ? self : typeof window < "u" ? window : typeof global < "u" ? global : {});
function et(e, ...t) {
  console.warn(`[Vue warn] ${e}`, ...t);
}
let En;
function Nn(e, t = En) {
  t && t.active && t.effects.push(e);
}
const ne = (e) => {
  const t = new Set(e);
  return t.w = 0, t.n = 0, t;
}, ht = (e) => (e.w & z) > 0, _t = (e) => (e.n & z) > 0, wn = ({ deps: e }) => {
  if (e.length)
    for (let t = 0; t < e.length; t++)
      e[t].w |= z;
}, bn = (e) => {
  const { deps: t } = e;
  if (t.length) {
    let n = 0;
    for (let s = 0; s < t.length; s++) {
      const r = t[s];
      ht(r) && !_t(r) ? r.delete(e) : t[n++] = r, r.w &= ~z, r.n &= ~z;
    }
    t.length = n;
  }
}, Re = /* @__PURE__ */ new WeakMap();
let k = 0, z = 1;
const De = 30;
let O;
const W = Symbol(process.env.NODE_ENV !== "production" ? "iterate" : ""), $e = Symbol(process.env.NODE_ENV !== "production" ? "Map key iterate" : "");
class gt {
  constructor(t, n = null, s) {
    this.fn = t, this.scheduler = n, this.active = !0, this.deps = [], this.parent = void 0, Nn(this, s);
  }
  run() {
    if (!this.active)
      return this.fn();
    let t = O, n = j;
    for (; t; ) {
      if (t === this)
        return;
      t = t.parent;
    }
    try {
      return this.parent = O, O = this, j = !0, z = 1 << ++k, k <= De ? wn(this) : tt(this), this.fn();
    } finally {
      k <= De && bn(this), z = 1 << --k, O = this.parent, j = n, this.parent = void 0, this.deferStop && this.stop();
    }
  }
  stop() {
    O === this ? this.deferStop = !0 : this.active && (tt(this), this.onStop && this.onStop(), this.active = !1);
  }
}
function tt(e) {
  const { deps: t } = e;
  if (t.length) {
    for (let n = 0; n < t.length; n++)
      t[n].delete(e);
    t.length = 0;
  }
}
let j = !0;
const mt = [];
function Et() {
  mt.push(j), j = !1;
}
function Nt() {
  const e = mt.pop();
  j = e === void 0 ? !0 : e;
}
function I(e, t, n) {
  if (j && O) {
    let s = Re.get(e);
    s || Re.set(e, s = /* @__PURE__ */ new Map());
    let r = s.get(n);
    r || s.set(n, r = ne());
    const o = process.env.NODE_ENV !== "production" ? { effect: O, target: e, type: t, key: n } : void 0;
    Ce(r, o);
  }
}
function Ce(e, t) {
  let n = !1;
  k <= De ? _t(e) || (e.n |= z, n = !ht(e)) : n = !e.has(O), n && (e.add(O), O.deps.push(e), process.env.NODE_ENV !== "production" && O.onTrack && O.onTrack(Object.assign({ effect: O }, t)));
}
function K(e, t, n, s, r, o) {
  const i = Re.get(e);
  if (!i)
    return;
  let c = [];
  if (t === "clear")
    c = [...i.values()];
  else if (n === "length" && h(e)) {
    const p = gn(s);
    i.forEach((d, l) => {
      (l === "length" || l >= p) && c.push(d);
    });
  } else
    switch (n !== void 0 && c.push(i.get(n)), t) {
      case "add":
        h(e) ? He(n) && c.push(i.get("length")) : (c.push(i.get(W)), Y(e) && c.push(i.get($e)));
        break;
      case "delete":
        h(e) || (c.push(i.get(W)), Y(e) && c.push(i.get($e)));
        break;
      case "set":
        Y(e) && c.push(i.get(W));
        break;
    }
  const u = process.env.NODE_ENV !== "production" ? { target: e, type: t, key: n, newValue: s, oldValue: r, oldTarget: o } : void 0;
  if (c.length === 1)
    c[0] && (process.env.NODE_ENV !== "production" ? q(c[0], u) : q(c[0]));
  else {
    const p = [];
    for (const d of c)
      d && p.push(...d);
    process.env.NODE_ENV !== "production" ? q(ne(p), u) : q(ne(p));
  }
}
function q(e, t) {
  const n = h(e) ? e : [...e];
  for (const s of n)
    s.computed && nt(s, t);
  for (const s of n)
    s.computed || nt(s, t);
}
function nt(e, t) {
  (e !== O || e.allowRecurse) && (process.env.NODE_ENV !== "production" && e.onTrigger && e.onTrigger($({ effect: e }, t)), e.scheduler ? e.scheduler() : e.run());
}
const On = /* @__PURE__ */ Zt("__proto__,__v_isRef,__isVue"), wt = new Set(
  /* @__PURE__ */ Object.getOwnPropertyNames(Symbol).filter((e) => e !== "arguments" && e !== "caller").map((e) => Symbol[e]).filter(Ke)
), Sn = /* @__PURE__ */ We(), Vn = /* @__PURE__ */ We(!0), yn = /* @__PURE__ */ We(!0, !0), rt = /* @__PURE__ */ vn();
function vn() {
  const e = {};
  return ["includes", "indexOf", "lastIndexOf"].forEach((t) => {
    e[t] = function(...n) {
      const s = f(this);
      for (let o = 0, i = this.length; o < i; o++)
        I(s, "get", o + "");
      const r = s[t](...n);
      return r === -1 || r === !1 ? s[t](...n.map(f)) : r;
    };
  }), ["push", "pop", "shift", "unshift", "splice"].forEach((t) => {
    e[t] = function(...n) {
      Et();
      const s = f(this)[t].apply(this, n);
      return Nt(), s;
    };
  }), e;
}
function We(e = !1, t = !1) {
  return function(s, r, o) {
    if (r === "__v_isReactive")
      return !e;
    if (r === "__v_isReadonly")
      return e;
    if (r === "__v_isShallow")
      return t;
    if (r === "__v_raw" && o === (e ? t ? yt : Vt : t ? Wn : St).get(s))
      return s;
    const i = h(s);
    if (!e && i && g(rt, r))
      return Reflect.get(rt, r, o);
    const c = Reflect.get(s, r, o);
    return (Ke(r) ? wt.has(r) : On(r)) || (e || I(s, "get", r), t) ? c : S(c) ? i && He(r) ? c : c.value : V(c) ? e ? xt(c) : vt(c) : c;
  };
}
const xn = /* @__PURE__ */ In();
function In(e = !1) {
  return function(n, s, r, o) {
    let i = n[s];
    if (J(i) && S(i) && !S(r))
      return !1;
    if (!e && (!Te(r) && !J(r) && (i = f(i), r = f(r)), !h(n) && S(i) && !S(r)))
      return i.value = r, !0;
    const c = h(n) && He(s) ? Number(s) < n.length : g(n, s), u = Reflect.set(n, s, r, o);
    return n === f(o) && (c ? ge(r, i) && K(n, "set", s, r, i) : K(n, "add", s, r)), u;
  };
}
function Rn(e, t) {
  const n = g(e, t), s = e[t], r = Reflect.deleteProperty(e, t);
  return r && n && K(e, "delete", t, void 0, s), r;
}
function Dn(e, t) {
  const n = Reflect.has(e, t);
  return (!Ke(t) || !wt.has(t)) && I(e, "has", t), n;
}
function $n(e) {
  return I(e, "iterate", h(e) ? "length" : W), Reflect.ownKeys(e);
}
const Cn = {
  get: Sn,
  set: xn,
  deleteProperty: Rn,
  has: Dn,
  ownKeys: $n
}, bt = {
  get: Vn,
  set(e, t) {
    return process.env.NODE_ENV !== "production" && et(`Set operation on key "${String(t)}" failed: target is readonly.`, e), !0;
  },
  deleteProperty(e, t) {
    return process.env.NODE_ENV !== "production" && et(`Delete operation on key "${String(t)}" failed: target is readonly.`, e), !0;
  }
}, Tn = /* @__PURE__ */ $({}, bt, {
  get: yn
}), Ue = (e) => e, Se = (e) => Reflect.getPrototypeOf(e);
function oe(e, t, n = !1, s = !1) {
  e = e.__v_raw;
  const r = f(e), o = f(t);
  n || (t !== o && I(r, "get", t), I(r, "get", o));
  const { has: i } = Se(r), c = s ? Ue : n ? qe : Je;
  if (i.call(r, t))
    return c(e.get(t));
  if (i.call(r, o))
    return c(e.get(o));
  e !== r && e.get(t);
}
function ie(e, t = !1) {
  const n = this.__v_raw, s = f(n), r = f(e);
  return t || (e !== r && I(s, "has", e), I(s, "has", r)), e === r ? n.has(e) : n.has(e) || n.has(r);
}
function ce(e, t = !1) {
  return e = e.__v_raw, !t && I(f(e), "iterate", W), Reflect.get(e, "size", e);
}
function st(e) {
  e = f(e);
  const t = f(this);
  return Se(t).has.call(t, e) || (t.add(e), K(t, "add", e, e)), this;
}
function ot(e, t) {
  t = f(t);
  const n = f(this), { has: s, get: r } = Se(n);
  let o = s.call(n, e);
  o ? process.env.NODE_ENV !== "production" && Ot(n, s, e) : (e = f(e), o = s.call(n, e));
  const i = r.call(n, e);
  return n.set(e, t), o ? ge(t, i) && K(n, "set", e, t, i) : K(n, "add", e, t), this;
}
function it(e) {
  const t = f(this), { has: n, get: s } = Se(t);
  let r = n.call(t, e);
  r ? process.env.NODE_ENV !== "production" && Ot(t, n, e) : (e = f(e), r = n.call(t, e));
  const o = s ? s.call(t, e) : void 0, i = t.delete(e);
  return r && K(t, "delete", e, void 0, o), i;
}
function ct() {
  const e = f(this), t = e.size !== 0, n = process.env.NODE_ENV !== "production" ? Y(e) ? new Map(e) : new Set(e) : void 0, s = e.clear();
  return t && K(e, "clear", void 0, void 0, n), s;
}
function le(e, t) {
  return function(s, r) {
    const o = this, i = o.__v_raw, c = f(i), u = t ? Ue : e ? qe : Je;
    return !e && I(c, "iterate", W), i.forEach((p, d) => s.call(r, u(p), u(d), o));
  };
}
function ue(e, t, n) {
  return function(...s) {
    const r = this.__v_raw, o = f(r), i = Y(o), c = e === "entries" || e === Symbol.iterator && i, u = e === "keys" && i, p = r[e](...s), d = n ? Ue : t ? qe : Je;
    return !t && I(o, "iterate", u ? $e : W), {
      next() {
        const { value: l, done: a } = p.next();
        return a ? { value: l, done: a } : {
          value: c ? [d(l[0]), d(l[1])] : d(l),
          done: a
        };
      },
      [Symbol.iterator]() {
        return this;
      }
    };
  };
}
function M(e) {
  return function(...t) {
    if (process.env.NODE_ENV !== "production") {
      const n = t[0] ? `on key "${t[0]}" ` : "";
      console.warn(`${hn(e)} operation ${n}failed: target is readonly.`, f(this));
    }
    return e === "delete" ? !1 : this;
  };
}
function Pn() {
  const e = {
    get(o) {
      return oe(this, o);
    },
    get size() {
      return ce(this);
    },
    has: ie,
    add: st,
    set: ot,
    delete: it,
    clear: ct,
    forEach: le(!1, !1)
  }, t = {
    get(o) {
      return oe(this, o, !1, !0);
    },
    get size() {
      return ce(this);
    },
    has: ie,
    add: st,
    set: ot,
    delete: it,
    clear: ct,
    forEach: le(!1, !0)
  }, n = {
    get(o) {
      return oe(this, o, !0);
    },
    get size() {
      return ce(this, !0);
    },
    has(o) {
      return ie.call(this, o, !0);
    },
    add: M("add"),
    set: M("set"),
    delete: M("delete"),
    clear: M("clear"),
    forEach: le(!0, !1)
  }, s = {
    get(o) {
      return oe(this, o, !0, !0);
    },
    get size() {
      return ce(this, !0);
    },
    has(o) {
      return ie.call(this, o, !0);
    },
    add: M("add"),
    set: M("set"),
    delete: M("delete"),
    clear: M("clear"),
    forEach: le(!0, !0)
  };
  return ["keys", "values", "entries", Symbol.iterator].forEach((o) => {
    e[o] = ue(o, !1, !1), n[o] = ue(o, !0, !1), t[o] = ue(o, !1, !0), s[o] = ue(o, !0, !0);
  }), [
    e,
    n,
    t,
    s
  ];
}
const [Mn, Fn, An, jn] = /* @__PURE__ */ Pn();
function Be(e, t) {
  const n = t ? e ? jn : An : e ? Fn : Mn;
  return (s, r, o) => r === "__v_isReactive" ? !e : r === "__v_isReadonly" ? e : r === "__v_raw" ? s : Reflect.get(g(n, r) && r in s ? n : s, r, o);
}
const zn = {
  get: /* @__PURE__ */ Be(!1, !1)
}, Kn = {
  get: /* @__PURE__ */ Be(!0, !1)
}, Hn = {
  get: /* @__PURE__ */ Be(!0, !0)
};
function Ot(e, t, n) {
  const s = f(n);
  if (s !== n && t.call(e, s)) {
    const r = dt(e);
    console.warn(`Reactive ${r} contains both the raw and reactive versions of the same object${r === "Map" ? " as keys" : ""}, which can lead to inconsistencies. Avoid differentiating between the raw and reactive versions of an object and only use the reactive version if possible.`);
  }
}
const St = /* @__PURE__ */ new WeakMap(), Wn = /* @__PURE__ */ new WeakMap(), Vt = /* @__PURE__ */ new WeakMap(), yt = /* @__PURE__ */ new WeakMap();
function Un(e) {
  switch (e) {
    case "Object":
    case "Array":
      return 1;
    case "Map":
    case "Set":
    case "WeakMap":
    case "WeakSet":
      return 2;
    default:
      return 0;
  }
}
function Bn(e) {
  return e.__v_skip || !Object.isExtensible(e) ? 0 : Un(dt(e));
}
function vt(e) {
  return J(e) ? e : Ge(e, !1, Cn, zn, St);
}
function xt(e) {
  return Ge(e, !0, bt, Kn, Vt);
}
function ae(e) {
  return Ge(e, !0, Tn, Hn, yt);
}
function Ge(e, t, n, s, r) {
  if (!V(e))
    return process.env.NODE_ENV !== "production" && console.warn(`value cannot be made reactive: ${String(e)}`), e;
  if (e.__v_raw && !(t && e.__v_isReactive))
    return e;
  const o = r.get(e);
  if (o)
    return o;
  const i = Bn(e);
  if (i === 0)
    return e;
  const c = new Proxy(e, i === 2 ? s : n);
  return r.set(e, c), c;
}
function U(e) {
  return J(e) ? U(e.__v_raw) : !!(e && e.__v_isReactive);
}
function J(e) {
  return !!(e && e.__v_isReadonly);
}
function Te(e) {
  return !!(e && e.__v_isShallow);
}
function Pe(e) {
  return U(e) || J(e);
}
function f(e) {
  const t = e && e.__v_raw;
  return t ? f(t) : e;
}
function Gn(e) {
  return _n(e, "__v_skip", !0), e;
}
const Je = (e) => V(e) ? vt(e) : e, qe = (e) => V(e) ? xt(e) : e;
function Jn(e) {
  j && O && (e = f(e), process.env.NODE_ENV !== "production" ? Ce(e.dep || (e.dep = ne()), {
    target: e,
    type: "get",
    key: "value"
  }) : Ce(e.dep || (e.dep = ne())));
}
function qn(e, t) {
  e = f(e), e.dep && (process.env.NODE_ENV !== "production" ? q(e.dep, {
    target: e,
    type: "set",
    key: "value",
    newValue: t
  }) : q(e.dep));
}
function S(e) {
  return !!(e && e.__v_isRef === !0);
}
function It(e) {
  return S(e) ? e.value : e;
}
const Ln = {
  get: (e, t, n) => It(Reflect.get(e, t, n)),
  set: (e, t, n, s) => {
    const r = e[t];
    return S(r) && !S(n) ? (r.value = n, !0) : Reflect.set(e, t, n, s);
  }
};
function Yn(e) {
  return U(e) ? e : new Proxy(e, Ln);
}
var Rt;
class Qn {
  constructor(t, n, s, r) {
    this._setter = n, this.dep = void 0, this.__v_isRef = !0, this[Rt] = !1, this._dirty = !0, this.effect = new gt(t, () => {
      this._dirty || (this._dirty = !0, qn(this));
    }), this.effect.computed = this, this.effect.active = this._cacheable = !r, this.__v_isReadonly = s;
  }
  get value() {
    const t = f(this);
    return Jn(t), (t._dirty || !t._cacheable) && (t._dirty = !1, t._value = t.effect.run()), t._value;
  }
  set value(t) {
    this._setter(t);
  }
}
Rt = "__v_isReadonly";
function Xn(e, t, n = !1) {
  let s, r;
  const o = w(e);
  o ? (s = e, r = process.env.NODE_ENV !== "production" ? () => {
    console.warn("Write operation failed: computed value is readonly");
  } : ze) : (s = e.get, r = e.set);
  const i = new Qn(s, r, o || !r, n);
  return process.env.NODE_ENV !== "production" && t && !n && (i.effect.onTrack = t.onTrack, i.effect.onTrigger = t.onTrigger), i;
}
const B = [];
function Zn(e) {
  B.push(e);
}
function kn() {
  B.pop();
}
function N(e, ...t) {
  if (process.env.NODE_ENV === "production")
    return;
  Et();
  const n = B.length ? B[B.length - 1].component : null, s = n && n.appContext.config.warnHandler, r = er();
  if (s)
    G(s, n, 11, [
      e + t.join(""),
      n && n.proxy,
      r.map(({ vnode: o }) => `at <${Lt(n, o.type)}>`).join(`
`),
      r
    ]);
  else {
    const o = [`[Vue warn]: ${e}`, ...t];
    r.length && o.push(`
`, ...tr(r)), console.warn(...o);
  }
  Nt();
}
function er() {
  let e = B[B.length - 1];
  if (!e)
    return [];
  const t = [];
  for (; e; ) {
    const n = t[0];
    n && n.vnode === e ? n.recurseCount++ : t.push({
      vnode: e,
      recurseCount: 0
    });
    const s = e.component && e.component.parent;
    e = s && s.vnode;
  }
  return t;
}
function tr(e) {
  const t = [];
  return e.forEach((n, s) => {
    t.push(...s === 0 ? [] : [`
`], ...nr(n));
  }), t;
}
function nr({ vnode: e, recurseCount: t }) {
  const n = t > 0 ? `... (${t} recursive calls)` : "", s = e.component ? e.component.parent == null : !1, r = ` at <${Lt(e.component, e.type, s)}`, o = ">" + n;
  return e.props ? [r, ...rr(e.props), o] : [r + o];
}
function rr(e) {
  const t = [], n = Object.keys(e);
  return n.slice(0, 3).forEach((s) => {
    t.push(...Dt(s, e[s]));
  }), n.length > 3 && t.push(" ..."), t;
}
function Dt(e, t, n) {
  return C(t) ? (t = JSON.stringify(t), n ? t : [`${e}=${t}`]) : typeof t == "number" || typeof t == "boolean" || t == null ? n ? t : [`${e}=${t}`] : S(t) ? (t = Dt(e, f(t.value), !0), n ? t : [`${e}=Ref<`, t, ">"]) : w(t) ? [`${e}=fn${t.name ? `<${t.name}>` : ""}`] : (t = f(t), n ? t : [`${e}=`, t]);
}
const $t = {
  sp: "serverPrefetch hook",
  bc: "beforeCreate hook",
  c: "created hook",
  bm: "beforeMount hook",
  m: "mounted hook",
  bu: "beforeUpdate hook",
  u: "updated",
  bum: "beforeUnmount hook",
  um: "unmounted hook",
  a: "activated hook",
  da: "deactivated hook",
  ec: "errorCaptured hook",
  rtc: "renderTracked hook",
  rtg: "renderTriggered hook",
  [0]: "setup function",
  [1]: "render function",
  [2]: "watcher getter",
  [3]: "watcher callback",
  [4]: "watcher cleanup function",
  [5]: "native event handler",
  [6]: "component event handler",
  [7]: "vnode hook",
  [8]: "directive hook",
  [9]: "transition hook",
  [10]: "app errorHandler",
  [11]: "app warnHandler",
  [12]: "ref function",
  [13]: "async component loader",
  [14]: "scheduler flush. This is likely a Vue internals bug. Please open an issue at https://new-issue.vuejs.org/?repo=vuejs/core"
};
function G(e, t, n, s) {
  let r;
  try {
    r = s ? e(...s) : e();
  } catch (o) {
    Ct(o, t, n);
  }
  return r;
}
function Me(e, t, n, s) {
  if (w(e)) {
    const o = G(e, t, n, s);
    return o && an(o) && o.catch((i) => {
      Ct(i, t, n);
    }), o;
  }
  const r = [];
  for (let o = 0; o < e.length; o++)
    r.push(Me(e[o], t, n, s));
  return r;
}
function Ct(e, t, n, s = !0) {
  const r = t ? t.vnode : null;
  if (t) {
    let o = t.parent;
    const i = t.proxy, c = process.env.NODE_ENV !== "production" ? $t[n] : n;
    for (; o; ) {
      const p = o.ec;
      if (p) {
        for (let d = 0; d < p.length; d++)
          if (p[d](e, i, c) === !1)
            return;
      }
      o = o.parent;
    }
    const u = t.appContext.config.errorHandler;
    if (u) {
      G(u, null, 10, [e, i, c]);
      return;
    }
  }
  sr(e, n, r, s);
}
function sr(e, t, n, s = !0) {
  if (process.env.NODE_ENV !== "production") {
    const r = $t[t];
    if (n && Zn(n), N(`Unhandled error${r ? ` during execution of ${r}` : ""}`), n && kn(), s)
      throw e;
    console.error(e);
  } else
    console.error(e);
}
let me = !1, Fe = !1;
const R = [];
let A = 0;
const Q = [];
let T = null, F = 0;
const Tt = /* @__PURE__ */ Promise.resolve();
let Le = null;
const or = 100;
function ir(e) {
  const t = Le || Tt;
  return e ? t.then(this ? e.bind(this) : e) : t;
}
function cr(e) {
  let t = A + 1, n = R.length;
  for (; t < n; ) {
    const s = t + n >>> 1;
    re(R[s]) < e ? t = s + 1 : n = s;
  }
  return t;
}
function Ye(e) {
  (!R.length || !R.includes(e, me && e.allowRecurse ? A + 1 : A)) && (e.id == null ? R.push(e) : R.splice(cr(e.id), 0, e), Pt());
}
function Pt() {
  !me && !Fe && (Fe = !0, Le = Tt.then(Ft));
}
function Mt(e) {
  h(e) ? Q.push(...e) : (!T || !T.includes(e, e.allowRecurse ? F + 1 : F)) && Q.push(e), Pt();
}
function lr(e) {
  if (Q.length) {
    const t = [...new Set(Q)];
    if (Q.length = 0, T) {
      T.push(...t);
      return;
    }
    for (T = t, process.env.NODE_ENV !== "production" && (e = e || /* @__PURE__ */ new Map()), T.sort((n, s) => re(n) - re(s)), F = 0; F < T.length; F++)
      process.env.NODE_ENV !== "production" && At(e, T[F]) || T[F]();
    T = null, F = 0;
  }
}
const re = (e) => e.id == null ? 1 / 0 : e.id, ur = (e, t) => {
  const n = re(e) - re(t);
  if (n === 0) {
    if (e.pre && !t.pre)
      return -1;
    if (t.pre && !e.pre)
      return 1;
  }
  return n;
};
function Ft(e) {
  Fe = !1, me = !0, process.env.NODE_ENV !== "production" && (e = e || /* @__PURE__ */ new Map()), R.sort(ur);
  const t = process.env.NODE_ENV !== "production" ? (n) => At(e, n) : ze;
  try {
    for (A = 0; A < R.length; A++) {
      const n = R[A];
      if (n && n.active !== !1) {
        if (process.env.NODE_ENV !== "production" && t(n))
          continue;
        G(n, null, 14);
      }
    }
  } finally {
    A = 0, R.length = 0, lr(e), me = !1, Le = null, (R.length || Q.length) && Ft(e);
  }
}
function At(e, t) {
  if (!e.has(t))
    e.set(t, 1);
  else {
    const n = e.get(t);
    if (n > or) {
      const s = t.ownerInstance, r = s && qt(s.type);
      return N(`Maximum recursive updates exceeded${r ? ` in component <${r}>` : ""}. This means you have a reactive effect that is mutating its own dependencies and thus recursively triggering itself. Possible sources include component template, render function, updated hook or watcher source function.`), !0;
    } else
      e.set(t, n + 1);
  }
}
const Z = /* @__PURE__ */ new Set();
process.env.NODE_ENV !== "production" && (mn().__VUE_HMR_RUNTIME__ = {
  createRecord: ve(ar),
  rerender: ve(fr),
  reload: ve(pr)
});
const Ee = /* @__PURE__ */ new Map();
function ar(e, t) {
  return Ee.has(e) ? !1 : (Ee.set(e, {
    initialDef: ee(t),
    instances: /* @__PURE__ */ new Set()
  }), !0);
}
function ee(e) {
  return Yt(e) ? e.__vccOpts : e;
}
function fr(e, t) {
  const n = Ee.get(e);
  n && (n.initialDef.render = t, [...n.instances].forEach((s) => {
    t && (s.render = t, ee(s.type).render = t), s.renderCache = [], s.update();
  }));
}
function pr(e, t) {
  const n = Ee.get(e);
  if (!n)
    return;
  t = ee(t), lt(n.initialDef, t);
  const s = [...n.instances];
  for (const r of s) {
    const o = ee(r.type);
    Z.has(o) || (o !== n.initialDef && lt(o, t), Z.add(o)), r.appContext.optionsCache.delete(r.type), r.ceReload ? (Z.add(o), r.ceReload(t.styles), Z.delete(o)) : r.parent ? Ye(r.parent.update) : r.appContext.reload ? r.appContext.reload() : typeof window < "u" ? window.location.reload() : console.warn("[HMR] Root or manually mounted instance modified. Full reload required.");
  }
  Mt(() => {
    for (const r of s)
      Z.delete(ee(r.type));
  });
}
function lt(e, t) {
  $(e, t);
  for (const n in e)
    n !== "__file" && !(n in t) && delete e[n];
}
function ve(e) {
  return (t, n) => {
    try {
      return e(t, n);
    } catch (s) {
      console.error(s), console.warn("[HMR] Something went wrong during Vue component hot-reload. Full reload required.");
    }
  };
}
let x = null, dr = null;
const hr = (e) => e.__isSuspense;
function _r(e, t) {
  t && t.pendingBranch ? h(e) ? t.effects.push(...e) : t.effects.push(e) : Mt(e);
}
const fe = {};
function gr(e, t, { immediate: n, deep: s, flush: r, onTrack: o, onTrigger: i } = P) {
  process.env.NODE_ENV !== "production" && !t && (n !== void 0 && N('watch() "immediate" option is only respected when using the watch(source, callback, options?) signature.'), s !== void 0 && N('watch() "deep" option is only respected when using the watch(source, callback, options?) signature.'));
  const c = (_) => {
    N("Invalid watch source: ", _, "A watch source can only be a getter/effect function, a ref, a reactive object, or an array of these types.");
  }, u = X;
  let p, d = !1, l = !1;
  if (S(e) ? (p = () => e.value, d = Te(e)) : U(e) ? (p = () => e, s = !0) : h(e) ? (l = !0, d = e.some((_) => U(_) || Te(_)), p = () => e.map((_) => {
    if (S(_))
      return _.value;
    if (U(_))
      return L(_);
    if (w(_))
      return G(_, u, 2);
    process.env.NODE_ENV !== "production" && c(_);
  })) : w(e) ? t ? p = () => G(e, u, 2) : p = () => {
    if (!(u && u.isUnmounted))
      return a && a(), Me(e, u, 3, [m]);
  } : (p = ze, process.env.NODE_ENV !== "production" && c(e)), t && s) {
    const _ = p;
    p = () => L(_());
  }
  let a, m = (_) => {
    a = v.onStop = () => {
      G(_, u, 4);
    };
  }, E = l ? new Array(e.length).fill(fe) : fe;
  const y = () => {
    if (v.active)
      if (t) {
        const _ = v.run();
        (s || d || (l ? _.some((Qt, Xt) => ge(Qt, E[Xt])) : ge(_, E))) && (a && a(), Me(t, u, 3, [
          _,
          E === fe ? void 0 : l && E[0] === fe ? [] : E,
          m
        ]), E = _);
      } else
        v.run();
  };
  y.allowRecurse = !!t;
  let se;
  r === "sync" ? se = y : r === "post" ? se = () => ft(y, u && u.suspense) : (y.pre = !0, u && (y.id = u.uid), se = () => Ye(y));
  const v = new gt(p, se);
  return process.env.NODE_ENV !== "production" && (v.onTrack = o, v.onTrigger = i), t ? n ? y() : E = v.run() : r === "post" ? ft(v.run.bind(v), u && u.suspense) : v.run(), () => {
    v.stop(), u && u.scope && cn(u.scope.effects, v);
  };
}
function mr(e, t, n) {
  const s = this.proxy, r = C(e) ? e.includes(".") ? Er(s, e) : () => s[e] : e.bind(s, s);
  let o;
  w(t) ? o = t : (o = t.handler, n = t);
  const i = X;
  pt(this);
  const c = gr(r, o.bind(s), n);
  return i ? pt(i) : Fr(), c;
}
function Er(e, t) {
  const n = t.split(".");
  return () => {
    let s = e;
    for (let r = 0; r < n.length && s; r++)
      s = s[n[r]];
    return s;
  };
}
function L(e, t) {
  if (!V(e) || e.__v_skip || (t = t || /* @__PURE__ */ new Set(), t.has(e)))
    return e;
  if (t.add(e), S(e))
    L(e.value, t);
  else if (h(e))
    for (let n = 0; n < e.length; n++)
      L(e[n], t);
  else if (un(e) || Y(e))
    e.forEach((n) => {
      L(n, t);
    });
  else if (pn(e))
    for (const n in e)
      L(e[n], t);
  return e;
}
const Nr = (e) => !!e.type.__asyncLoader, wr = Symbol();
function br(e, t, n = {}, s, r) {
  if (x.isCE || x.parent && Nr(x.parent) && x.parent.isCE)
    return t !== "default" && (n.name = t), Xe("slot", n, s && s());
  let o = e[t];
  process.env.NODE_ENV !== "production" && o && o.length > 1 && (N("SSR-optimized slot function detected in a non-SSR-optimized render function. You need to mark this component with $dynamic-slots in the parent template."), o = () => []), o && o._c && (o._d = !1), Qe();
  const i = o && jt(o(n)), c = $r(Ve, {
    key: n.key || i && i.key || `_${t}`
  }, i || (s ? s() : []), i && e._ === 1 ? 64 : -2);
  return !r && c.scopeId && (c.slotScopeIds = [c.scopeId + "-s"]), o && o._c && (o._d = !0), c;
}
function jt(e) {
  return e.some((t) => Wt(t) ? !(t.type === zt || t.type === Ve && !jt(t.children)) : !0) ? e : null;
}
const Ae = (e) => e ? Ar(e) ? zr(e) || e.proxy : Ae(e.parent) : null, te = /* @__PURE__ */ $(/* @__PURE__ */ Object.create(null), {
  $: (e) => e,
  $el: (e) => e.vnode.el,
  $data: (e) => e.data,
  $props: (e) => process.env.NODE_ENV !== "production" ? ae(e.props) : e.props,
  $attrs: (e) => process.env.NODE_ENV !== "production" ? ae(e.attrs) : e.attrs,
  $slots: (e) => process.env.NODE_ENV !== "production" ? ae(e.slots) : e.slots,
  $refs: (e) => process.env.NODE_ENV !== "production" ? ae(e.refs) : e.refs,
  $parent: (e) => Ae(e.parent),
  $root: (e) => Ae(e.root),
  $emit: (e) => e.emit,
  $options: (e) => Vr(e),
  $forceUpdate: (e) => e.f || (e.f = () => Ye(e.update)),
  $nextTick: (e) => e.n || (e.n = ir.bind(e.proxy)),
  $watch: (e) => mr.bind(e)
}), Or = (e) => e === "_" || e === "$", xe = (e, t) => e !== P && !e.__isScriptSetup && g(e, t), Sr = {
  get({ _: e }, t) {
    const { ctx: n, setupState: s, data: r, props: o, accessCache: i, type: c, appContext: u } = e;
    if (process.env.NODE_ENV !== "production" && t === "__isVue")
      return !0;
    let p;
    if (t[0] !== "$") {
      const m = i[t];
      if (m !== void 0)
        switch (m) {
          case 1:
            return s[t];
          case 2:
            return r[t];
          case 4:
            return n[t];
          case 3:
            return o[t];
        }
      else {
        if (xe(s, t))
          return i[t] = 1, s[t];
        if (r !== P && g(r, t))
          return i[t] = 2, r[t];
        if ((p = e.propsOptions[0]) && g(p, t))
          return i[t] = 3, o[t];
        if (n !== P && g(n, t))
          return i[t] = 4, n[t];
        i[t] = 0;
      }
    }
    const d = te[t];
    let l, a;
    if (d)
      return t === "$attrs" && (I(e, "get", t), process.env.NODE_ENV !== "production" && void 0), d(e);
    if ((l = c.__cssModules) && (l = l[t]))
      return l;
    if (n !== P && g(n, t))
      return i[t] = 4, n[t];
    if (a = u.config.globalProperties, g(a, t))
      return a[t];
    process.env.NODE_ENV !== "production" && x && (!C(t) || t.indexOf("__v") !== 0) && (r !== P && Or(t[0]) && g(r, t) ? N(`Property ${JSON.stringify(t)} must be accessed via $data because it starts with a reserved character ("$" or "_") and is not proxied on the render context.`) : e === x && N(`Property ${JSON.stringify(t)} was accessed during render but is not defined on instance.`));
  },
  set({ _: e }, t, n) {
    const { data: s, setupState: r, ctx: o } = e;
    return xe(r, t) ? (r[t] = n, !0) : process.env.NODE_ENV !== "production" && r.__isScriptSetup && g(r, t) ? (N(`Cannot mutate <script setup> binding "${t}" from Options API.`), !1) : s !== P && g(s, t) ? (s[t] = n, !0) : g(e.props, t) ? (process.env.NODE_ENV !== "production" && N(`Attempting to mutate prop "${t}". Props are readonly.`), !1) : t[0] === "$" && t.slice(1) in e ? (process.env.NODE_ENV !== "production" && N(`Attempting to mutate public property "${t}". Properties starting with $ are reserved and readonly.`), !1) : (process.env.NODE_ENV !== "production" && t in e.appContext.config.globalProperties ? Object.defineProperty(o, t, {
      enumerable: !0,
      configurable: !0,
      value: n
    }) : o[t] = n, !0);
  },
  has({ _: { data: e, setupState: t, accessCache: n, ctx: s, appContext: r, propsOptions: o } }, i) {
    let c;
    return !!n[i] || e !== P && g(e, i) || xe(t, i) || (c = o[0]) && g(c, i) || g(s, i) || g(te, i) || g(r.config.globalProperties, i);
  },
  defineProperty(e, t, n) {
    return n.get != null ? e._.accessCache[t] = 0 : g(n, "value") && this.set(e, t, n.value, null), Reflect.defineProperty(e, t, n);
  }
};
process.env.NODE_ENV !== "production" && (Sr.ownKeys = (e) => (N("Avoid app logic that relies on enumerating keys on a component instance. The keys will be empty in production mode to avoid performance overhead."), Reflect.ownKeys(e)));
function Vr(e) {
  const t = e.type, { mixins: n, extends: s } = t, { mixins: r, optionsCache: o, config: { optionMergeStrategies: i } } = e.appContext, c = o.get(t);
  let u;
  return c ? u = c : !r.length && !n && !s ? u = t : (u = {}, r.length && r.forEach((p) => Ne(u, p, i, !0)), Ne(u, t, i)), V(t) && o.set(t, u), u;
}
function Ne(e, t, n, s = !1) {
  const { mixins: r, extends: o } = t;
  o && Ne(e, o, n, !0), r && r.forEach((i) => Ne(e, i, n, !0));
  for (const i in t)
    if (s && i === "expose")
      process.env.NODE_ENV !== "production" && N('"expose" option is ignored when declared in mixins or extends. It should only be declared in the base component itself.');
    else {
      const c = yr[i] || n && n[i];
      e[i] = c ? c(e[i], t[i]) : t[i];
    }
  return e;
}
const yr = {
  data: ut,
  props: H,
  emits: H,
  methods: H,
  computed: H,
  beforeCreate: b,
  created: b,
  beforeMount: b,
  mounted: b,
  beforeUpdate: b,
  updated: b,
  beforeDestroy: b,
  beforeUnmount: b,
  destroyed: b,
  unmounted: b,
  activated: b,
  deactivated: b,
  errorCaptured: b,
  serverPrefetch: b,
  components: H,
  directives: H,
  watch: xr,
  provide: ut,
  inject: vr
};
function ut(e, t) {
  return t ? e ? function() {
    return $(w(e) ? e.call(this, this) : e, w(t) ? t.call(this, this) : t);
  } : t : e;
}
function vr(e, t) {
  return H(at(e), at(t));
}
function at(e) {
  if (h(e)) {
    const t = {};
    for (let n = 0; n < e.length; n++)
      t[e[n]] = e[n];
    return t;
  }
  return e;
}
function b(e, t) {
  return e ? [...new Set([].concat(e, t))] : t;
}
function H(e, t) {
  return e ? $($(/* @__PURE__ */ Object.create(null), e), t) : t;
}
function xr(e, t) {
  if (!e)
    return t;
  if (!t)
    return e;
  const n = $(/* @__PURE__ */ Object.create(null), e);
  for (const s in t)
    n[s] = b(e[s], t[s]);
  return n;
}
const ft = _r, Ir = (e) => e.__isTeleport, Ve = Symbol(process.env.NODE_ENV !== "production" ? "Fragment" : void 0), Rr = Symbol(process.env.NODE_ENV !== "production" ? "Text" : void 0), zt = Symbol(process.env.NODE_ENV !== "production" ? "Comment" : void 0);
Symbol(process.env.NODE_ENV !== "production" ? "Static" : void 0);
const pe = [];
let D = null;
function Qe(e = !1) {
  pe.push(D = e ? null : []);
}
function Dr() {
  pe.pop(), D = pe[pe.length - 1] || null;
}
function Kt(e) {
  return e.dynamicChildren = D || rn, Dr(), D && D.push(e), e;
}
function Ht(e, t, n, s, r, o) {
  return Kt(ye(e, t, n, s, r, o, !0));
}
function $r(e, t, n, s, r) {
  return Kt(Xe(e, t, n, s, r, !0));
}
function Wt(e) {
  return e ? e.__v_isVNode === !0 : !1;
}
const Cr = (...e) => Gt(...e), Ut = "__vInternal", Bt = ({ key: e }) => e ?? null, de = ({ ref: e, ref_key: t, ref_for: n }) => e != null ? C(e) || S(e) || w(e) ? { i: x, r: e, k: t, f: !!n } : e : null;
function ye(e, t = null, n = null, s = 0, r = null, o = e === Ve ? 0 : 1, i = !1, c = !1) {
  const u = {
    __v_isVNode: !0,
    __v_skip: !0,
    type: e,
    props: t,
    key: t && Bt(t),
    ref: t && de(t),
    scopeId: dr,
    slotScopeIds: null,
    children: n,
    component: null,
    suspense: null,
    ssContent: null,
    ssFallback: null,
    dirs: null,
    transition: null,
    el: null,
    anchor: null,
    target: null,
    targetAnchor: null,
    staticCount: 0,
    shapeFlag: o,
    patchFlag: s,
    dynamicProps: r,
    dynamicChildren: null,
    appContext: null,
    ctx: x
  };
  return c ? (Ze(u, n), o & 128 && e.normalize(u)) : n && (u.shapeFlag |= C(n) ? 8 : 16), process.env.NODE_ENV !== "production" && u.key !== u.key && N("VNode created with invalid key (NaN). VNode type:", u.type), !i && D && (u.patchFlag > 0 || o & 6) && u.patchFlag !== 32 && D.push(u), u;
}
const Xe = process.env.NODE_ENV !== "production" ? Cr : Gt;
function Gt(e, t = null, n = null, s = 0, r = null, o = !1) {
  if ((!e || e === wr) && (process.env.NODE_ENV !== "production" && !e && N(`Invalid vnode type when creating vnode: ${e}.`), e = zt), Wt(e)) {
    const c = we(e, t, !0);
    return n && Ze(c, n), !o && D && (c.shapeFlag & 6 ? D[D.indexOf(e)] = c : D.push(c)), c.patchFlag |= -2, c;
  }
  if (Yt(e) && (e = e.__vccOpts), t) {
    t = Tr(t);
    let { class: c, style: u } = t;
    c && !C(c) && (t.class = be(c)), V(u) && (Pe(u) && !h(u) && (u = $({}, u)), t.style = je(u));
  }
  const i = C(e) ? 1 : hr(e) ? 128 : Ir(e) ? 64 : V(e) ? 4 : w(e) ? 2 : 0;
  return process.env.NODE_ENV !== "production" && i & 4 && Pe(e) && (e = f(e), N("Vue received a Component which was made a reactive object. This can lead to unnecessary performance overhead, and should be avoided by marking the component with `markRaw` or using `shallowRef` instead of `ref`.", `
Component that was made reactive: `, e)), ye(e, t, n, s, r, i, o, !0);
}
function Tr(e) {
  return e ? Pe(e) || Ut in e ? $({}, e) : e : null;
}
function we(e, t, n = !1) {
  const { props: s, ref: r, patchFlag: o, children: i } = e, c = t ? Mr(s || {}, t) : s;
  return {
    __v_isVNode: !0,
    __v_skip: !0,
    type: e.type,
    props: c,
    key: c && Bt(c),
    ref: t && t.ref ? n && r ? h(r) ? r.concat(de(t)) : [r, de(t)] : de(t) : r,
    scopeId: e.scopeId,
    slotScopeIds: e.slotScopeIds,
    children: process.env.NODE_ENV !== "production" && o === -1 && h(i) ? i.map(Jt) : i,
    target: e.target,
    targetAnchor: e.targetAnchor,
    staticCount: e.staticCount,
    shapeFlag: e.shapeFlag,
    patchFlag: t && e.type !== Ve ? o === -1 ? 16 : o | 16 : o,
    dynamicProps: e.dynamicProps,
    dynamicChildren: e.dynamicChildren,
    appContext: e.appContext,
    dirs: e.dirs,
    transition: e.transition,
    component: e.component,
    suspense: e.suspense,
    ssContent: e.ssContent && we(e.ssContent),
    ssFallback: e.ssFallback && we(e.ssFallback),
    el: e.el,
    anchor: e.anchor,
    ctx: e.ctx
  };
}
function Jt(e) {
  const t = we(e);
  return h(e.children) && (t.children = e.children.map(Jt)), t;
}
function Pr(e = " ", t = 0) {
  return Xe(Rr, null, e, t);
}
function Ze(e, t) {
  let n = 0;
  const { shapeFlag: s } = e;
  if (t == null)
    t = null;
  else if (h(t))
    n = 16;
  else if (typeof t == "object")
    if (s & 65) {
      const r = t.default;
      r && (r._c && (r._d = !1), Ze(e, r()), r._c && (r._d = !0));
      return;
    } else {
      n = 32;
      const r = t._;
      !r && !(Ut in t) ? t._ctx = x : r === 3 && x && (x.slots._ === 1 ? t._ = 1 : (t._ = 2, e.patchFlag |= 1024));
    }
  else
    w(t) ? (t = { default: t, _ctx: x }, n = 32) : (t = String(t), s & 64 ? (n = 16, t = [Pr(t)]) : n = 8);
  e.children = t, e.shapeFlag |= n;
}
function Mr(...e) {
  const t = {};
  for (let n = 0; n < e.length; n++) {
    const s = e[n];
    for (const r in s)
      if (r === "class")
        t.class !== s.class && (t.class = be([t.class, s.class]));
      else if (r === "style")
        t.style = je([t.style, s.style]);
      else if (on(r)) {
        const o = t[r], i = s[r];
        i && o !== i && !(h(o) && o.includes(i)) && (t[r] = o ? [].concat(o, i) : i);
      } else
        r !== "" && (t[r] = s[r]);
  }
  return t;
}
let X = null;
const pt = (e) => {
  X = e, e.scope.on();
}, Fr = () => {
  X && X.scope.off(), X = null;
};
function Ar(e) {
  return e.vnode.shapeFlag & 4;
}
let jr = !1;
function zr(e) {
  if (e.exposed)
    return e.exposeProxy || (e.exposeProxy = new Proxy(Yn(Gn(e.exposed)), {
      get(t, n) {
        if (n in t)
          return t[n];
        if (n in te)
          return te[n](e);
      },
      has(t, n) {
        return n in t || n in te;
      }
    }));
}
const Kr = /(?:^|[-_])(\w)/g, Hr = (e) => e.replace(Kr, (t) => t.toUpperCase()).replace(/[-_]/g, "");
function qt(e, t = !0) {
  return w(e) ? e.displayName || e.name : e.name || t && e.__name;
}
function Lt(e, t, n = !1) {
  let s = qt(t);
  if (!s && t.__file) {
    const r = t.__file.match(/([^/\\]+)\.\w+$/);
    r && (s = r[1]);
  }
  if (!s && e && e.parent) {
    const r = (o) => {
      for (const i in o)
        if (o[i] === t)
          return i;
    };
    s = r(e.components || e.parent.type.components) || r(e.appContext.components);
  }
  return s ? Hr(s) : n ? "App" : "Anonymous";
}
function Yt(e) {
  return w(e) && "__vccOpts" in e;
}
const Wr = (e, t) => Xn(e, t, jr);
Symbol(process.env.NODE_ENV !== "production" ? "ssrContext" : "");
function Ie(e) {
  return !!(e && e.__v_isShallow);
}
function Ur() {
  if (process.env.NODE_ENV === "production" || typeof window > "u")
    return;
  const e = { style: "color:#3ba776" }, t = { style: "color:#0b1bc9" }, n = { style: "color:#b62e24" }, s = { style: "color:#9d288c" }, r = {
    header(l) {
      return V(l) ? l.__isVue ? ["div", e, "VueInstance"] : S(l) ? [
        "div",
        {},
        ["span", e, d(l)],
        "<",
        c(l.value),
        ">"
      ] : U(l) ? [
        "div",
        {},
        ["span", e, Ie(l) ? "ShallowReactive" : "Reactive"],
        "<",
        c(l),
        `>${J(l) ? " (readonly)" : ""}`
      ] : J(l) ? [
        "div",
        {},
        ["span", e, Ie(l) ? "ShallowReadonly" : "Readonly"],
        "<",
        c(l),
        ">"
      ] : null : null;
    },
    hasBody(l) {
      return l && l.__isVue;
    },
    body(l) {
      if (l && l.__isVue)
        return [
          "div",
          {},
          ...o(l.$)
        ];
    }
  };
  function o(l) {
    const a = [];
    l.type.props && l.props && a.push(i("props", f(l.props))), l.setupState !== P && a.push(i("setup", l.setupState)), l.data !== P && a.push(i("data", f(l.data)));
    const m = u(l, "computed");
    m && a.push(i("computed", m));
    const E = u(l, "inject");
    return E && a.push(i("injected", E)), a.push([
      "div",
      {},
      [
        "span",
        {
          style: s.style + ";opacity:0.66"
        },
        "$ (internal): "
      ],
      ["object", { object: l }]
    ]), a;
  }
  function i(l, a) {
    return a = $({}, a), Object.keys(a).length ? [
      "div",
      { style: "line-height:1.25em;margin-bottom:0.6em" },
      [
        "div",
        {
          style: "color:#476582"
        },
        l
      ],
      [
        "div",
        {
          style: "padding-left:1.25em"
        },
        ...Object.keys(a).map((m) => [
          "div",
          {},
          ["span", s, m + ": "],
          c(a[m], !1)
        ])
      ]
    ] : ["span", {}];
  }
  function c(l, a = !0) {
    return typeof l == "number" ? ["span", t, l] : typeof l == "string" ? ["span", n, JSON.stringify(l)] : typeof l == "boolean" ? ["span", s, l] : V(l) ? ["object", { object: a ? f(l) : l }] : ["span", n, String(l)];
  }
  function u(l, a) {
    const m = l.type;
    if (w(m))
      return;
    const E = {};
    for (const y in l.ctx)
      p(m, y, a) && (E[y] = l.ctx[y]);
    return E;
  }
  function p(l, a, m) {
    const E = l[m];
    if (h(E) && E.includes(a) || V(E) && a in E || l.extends && p(l.extends, a, m) || l.mixins && l.mixins.some((y) => p(y, a, m)))
      return !0;
  }
  function d(l) {
    return Ie(l) ? "ShallowRef" : l.effect ? "ComputedRef" : "Ref";
  }
  window.devtoolsFormatters ? window.devtoolsFormatters.push(r) : window.devtoolsFormatters = [r];
}
function Br() {
  Ur();
}
process.env.NODE_ENV !== "production" && Br();
const Gr = (e, t) => {
  const n = e.__vccOpts || e;
  for (const [s, r] of t)
    n[s] = r;
  return n;
}, Jr = {
  name: "cButton"
}, qr = /* @__PURE__ */ Object.assign(Jr, {
  props: {
    type: {
      type: String,
      default: "default"
    }
  },
  setup(e) {
    const t = e, n = Wr(() => ["c-button", `c-button-${t.type}`]);
    return (s, r) => (Qe(), Ht("div", {
      class: be(It(n))
    }, [
      ye("button", null, [
        br(s.$slots, "default", {}, void 0, !0)
      ])
    ], 2));
  }
}), he = /* @__PURE__ */ Gr(qr, [["__scopeId", "data-v-996ee5e6"]]);
he.install = (e) => {
  e.component(he.name, he);
};
const Lr = ["value"], Yr = {
  name: "cInput"
}, _e = /* @__PURE__ */ Object.assign(Yr, {
  props: {
    modelValue: String | Number
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const n = (s) => {
      t("update:modelValue", s.target.value);
    };
    return (s, r) => (Qe(), Ht("div", null, [
      ye("input", {
        type: "text",
        value: e.modelValue,
        onInput: n
      }, null, 40, Lr)
    ]));
  }
});
_e.install = (e) => {
  e.component(_e.name, _e);
};
const Qr = (e) => {
  e.use(he), e.use(_e);
}, Zr = {
  install: Qr
};
export {
  he as cButton,
  _e as cInput,
  Zr as default
};
